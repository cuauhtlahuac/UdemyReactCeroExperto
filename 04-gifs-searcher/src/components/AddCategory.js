import React, { useState } from 'react';
import PropTypes from 'prop-types';

function AddCategory({
	setCategories,
	setCategoryDictionary,
	setSameValue,
	categoryDictionary,
}) {
	const [ inputValue, setInputValue ] = useState('');

	const handleInputValue = e => setInputValue(e.target.value);

	const handleSubmit = e => {
		e.preventDefault();

		const trimmedInputValue = inputValue.trim();

		if (trimmedInputValue && trimmedInputValue.length > 1) {
			setCategoryDictionary(catDic => ({
				...catDic,
				[trimmedInputValue]: trimmedInputValue,
			}));

			if (categoryDictionary.hasOwnProperty(trimmedInputValue)) {
				setSameValue(true);
			} else {
				setCategories(cat => [ trimmedInputValue, ...cat ]);
			}
			setInputValue('');
		}
	};

	return (
		<form onSubmit={handleSubmit} className="form">
		<label htmlFor={"category"}>Category</label>
			<input
				type="text"
				id={"category"}
				value={inputValue}
				placeholder="write a favorite gif category"
				onChange={handleInputValue}
			/>
		</form>
	);
}

export default AddCategory;

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};
