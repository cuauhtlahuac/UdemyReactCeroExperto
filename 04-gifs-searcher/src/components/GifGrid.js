import React, { lazy, Suspense } from 'react';
import PropTypes from 'prop-types';

import { useGetGifs } from 'hooks/useGetGifs';

const GifGridItem = lazy(() => import('./GifGridItem'));

const Loader = () => <div className="loading">Loading...</div>;

const GifGrid = ({ category }) => {
	const { data: images, loading } = useGetGifs(category);

	return (
		<section>
			<h3 className=
			"title animate__animated
			 animate__slideInRight"
			 >
				{category}
			</h3>

			{loading && (
				<p className=
				"loading animate__animated
				animate__fadeIn animate__repeat-3
				animate__faster"
				>
					loading...
				</p>
			)}

			<section className="image-grid">
				{images.map(image => (
					<Suspense fallback={ <Loader /> } key={image.id}>
						<GifGridItem {...image} />
					</Suspense>
				))}
			</section>
		</section>
	);
};

export default GifGrid;

GifGrid.propTypes = {
	category: PropTypes.string,
};
