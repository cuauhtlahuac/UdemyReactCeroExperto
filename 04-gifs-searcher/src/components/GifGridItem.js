import React, { useState, useEffect } from 'react';
import PropTypes from "prop-types";

export const GifGridItem = ({ gif, title }) => {
	const [ imageLoaded, setImageLoaded ] = useState(false);
	const [ clicked, setClicked ] = useState(false);

	useEffect(() => {}, [ gif, imageLoaded ]);
	return (
		<article 
			className={`card animate__animated 
			animate__flipInX
			${clicked && 'one-space'}`}
			onClick={()=> setClicked(!clicked)}
		>
			<img
				src={gif}
				alt={title}
				onLoad={() => setImageLoaded(true)}
				className={`smooth-image image-${imageLoaded
				? 'visible'
				: 'hidden'} ${(!title || title === ' ') && 'border-radius-bottom'}`}
			/>

			{(title && title !== ' ') && <p>{title}</p>}
		</article>
	);
};

export default React.memo(GifGridItem);
