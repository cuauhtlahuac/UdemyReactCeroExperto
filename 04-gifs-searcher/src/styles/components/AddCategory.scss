form {
	font-family: 'open sans', Arial, Helvetica, sans-serif;
	display: flex;
	flex-direction: column;
	margin-top: 1.3rem;
}

form > label {
	margin-bottom: 5px;

animation-duration: 5s;
animation-iteration-count: infinite;
animation-name: label-color;
}

@keyframes label-color {
	from {
		color: white;
	}

	to {
		color: black;
	}
}