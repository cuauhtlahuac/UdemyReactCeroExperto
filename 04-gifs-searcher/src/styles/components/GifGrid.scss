.image-grid {
	align-items: baseline;
	display: grid;
	grid-gap: 10px;
    grid-template-columns: repeat(auto-fit, minmax(16em, 1fr));
    row-gap: 15px;
	min-width: 0;
}

.loading {
	font-size: 2rem;
	font-family: 'Rubik' Arial, Helvetica, sans-serif;
	font-weight: 500;
	color: rgba(0, 0, 0, 0.274);
}