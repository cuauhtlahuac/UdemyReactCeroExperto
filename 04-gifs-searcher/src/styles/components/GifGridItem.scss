.card:hover {
	border: teal 2px outset;
	cursor: pointer;
}

.card {
	align-self: flex-start;
	background-color: whitesmoke;
	border: #ccc solid 1px;
	border-radius: 1rem;
	box-shadow: #ddd 1px 1px 10px;
	display: grid;
	justify-items: center;
    font-size: 1.2rem;
    width: auto;
}

.card p {
	color: darkslategray;
	padding: 0 1rem;
    text-align: center;
}

.one-space {
	animation-duration: 1s;
	animation-name: one-space;
}

@keyframes one-space {
	from {
		z-index: 0;
	}

	to {
        width: 200%;
		z-index: 2;
	}
}

.none-space {
	animation-duration: 1s;
	animation-name: one-space;
}

img {
	border-top-left-radius: 1rem;
	border-top-right-radius: 1rem;
    width: 100%;
    box-shadow: 0 0 0 1px #000;
    object-fit: cover;
}

.border-radius-bottom {
	border-bottom-left-radius: 1rem;
	border-bottom-right-radius: 1rem;
}

.smooth-image {
	transition: opacity 1s;
}

.image-visible {
	opacity: 1;
}

.image-hidden {
	opacity: 0.13;
}