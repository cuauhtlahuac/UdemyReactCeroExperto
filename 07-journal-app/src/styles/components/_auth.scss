.auth__main {
	align-items: center;
	background-color: $primary; // no es necesario importar la variable
	display: flex;
	justify-content: center;
	width: 100vw;
	height: 100vh;
}

.auth__box-container {
	background-color: white;
	box-shadow: 0 3px $dark-gray;
	border-radius: 2px;
	padding: 20px;
	width: 250px;
}

.auth__title {
	color: darken(
		$color: $primary,
		$amount: 15
	); // funciÃ³n de sass que obscurece el color
}

.auth__input {
	color: $dark-gray;
	border: 0;
	border-bottom: 1px solid $light-gray;
	font-size: 16px;
	margin-bottom: 30px;
	height: 20px;
	width: 100%;

	transition: border-bottom .3s ease;

	&:focus {
		outline: none;
		border-bottom: solid 2px $primary;
	}
}

.auth__social-networks {
	align-items: center;
	display: flex;
	justify-content: center;
	flex-direction: column;
	margin-bottom: 20px;
	width: 100%;
}

.auth_alert_error {
	background-color: lighten(desaturate($alert, 5), 15);
	border-radius: 5px;
	color: lighten($alert, $amount: 48);
	display: flex;
	justify-content: center;
	margin-bottom: 10px;
	transition: margin 1s ease;
	& > span {
		animation-duration: .5s;
		animation-name: animate-error;
		margin: 10px;
		opacity: 1;
	}
}

@keyframes animate-error {
	from {
		margin: 0px;
		opacity: 0;
	}
}

label {
	color: $dark-gray;
	&:hover {
		transition: margin-bottom 10 1s ease;
		transition: border-top .3s ease;
	}
}
