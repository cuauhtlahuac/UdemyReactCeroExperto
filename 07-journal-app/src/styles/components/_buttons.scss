$border-width: .15rem;
@mixin background-button-color($hue){
	background-color: adjust-color(
					$color: invert($primary),
					$hue: $hue,
					$saturation: 50%,
					$lightness: 10%,
					$alpha: 1.0
				);
}


.btn {
	background-color: transparent;
	border: none;
	color: white;
	cursor: pointer;
	font-size: 12px;
	padding: 7px 10px;

	transition: color .3s ease;

	&.btn-active:not(:disabled) {
		@keyframes changeBgc {
			from {
				@include background-button-color(300);
				border: $border-width white solid;
				color: white;
			}
			to {
				@include background-button-color(0);
				color: darken($primary, 30);
				border: $border-width invert($color: white) solid;
			}
		}

		border-radius: .3rem;
		background-color: darken($color: $primary, $amount: 10);
		transition-property: border color;
		transition-duration: 1s;

		& {
			animation-name: changeBgc;
			animation-duration: 3s;
			animation-iteration-count: infinite;
			animation-direction: alternate;
		}
	}

	&.btn-active {
		&:disabled {
			color: darken($primary, 10);
			border: $border-width darken($primary, 10) solid;
			transition-property: color;
			transition-duration: 1s;
		}

		&:focus {
			outline: none;
		}
	}
	&:hover:not(:disabled) {
		color: darken(white, 30);
	}
}

.btn-primary {
	background-color: $primary;
	border-radius: 2px;

	&:disabled {
		background-color: lighten($color: $primary, $amount: 20);
		color: lighten($color: $primary, $amount: 20);
	}

	&:hover {
		background-color: darken($color: $primary, $amount: 15);
	}
}

.btn-danger {
	background-color: red;
	height: 60px;
	transition: background-color .3s ease;
	&:hover{
		background-color: darken($color: red, $amount: 10);
	}
}

.btn-block {
	width: 100%;
}

.google-btn {
	cursor: pointer;
	margin-top: 5px;
	width: 100%;
	height: 42px;
	background-color: $google-blue;
	border-radius: 2px;
	box-shadow: 0 3px 4px 0 rgba(0, 0, 0, 0.25);

	transition: box-shadow .3s ease;

	.google-icon-wrapper {
		position: absolute;
		margin-top: 1px;
		margin-left: 1px;
		width: 40px;
		height: 40px;
		border-radius: 2px;
		background-color: $white;
	}
	.google-icon {
		position: absolute;
		margin-top: 11px;
		margin-left: 11px;
		width: 18px;
		height: 18px;
	}
	.btn-text {
		float: right;
		margin: 11px 40px 0 0;
		color: $white;
		font-size: 14px;
		letter-spacing: 0.2px;
	}
	&:hover {
		box-shadow: 0 0 6px $google-blue;
	}
	&:active {
		background: $button-active-blue;
	}
}
