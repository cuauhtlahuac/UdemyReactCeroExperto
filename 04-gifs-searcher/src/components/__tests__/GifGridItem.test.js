import React from 'react';
import { shallow } from 'enzyme';

import { GifGridItem } from 'components/GifGridItem';


describe('<GifGridItem />', () => {

    test('Should render correctly', () => {
        const component = shallow(<GifGridItem />);

        expect( component ).toMatchSnapshot();    
    })
    
})
