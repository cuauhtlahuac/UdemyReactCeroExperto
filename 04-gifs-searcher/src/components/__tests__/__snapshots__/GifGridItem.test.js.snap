// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> Should render correctly 1`] = `
<article
  className="card animate__animated 
			animate__flipInX
			false"
  onClick={[Function]}
>
  <img
    className="smooth-image image-hidden border-radius-bottom"
    onLoad={[Function]}
  />
</article>
`;
